### ISSUE TITLE
Dueling + Double DQN — 3 problemy architektoniczne

### ISSUE THREAD

**Issue by ArgonZapan** [ArgonZapan/DQN-transformer#31]
## Kontekst

Implementacja Dueling DQN i Double DQN jest logicznie poprawna — formuły, separacja wyboru/oceny akcji i action masking działają prawidłowo. Poniżej 3 problemy architektoniczne które obniżają jakość treningu.

---

## Problem 1 — BatchNorm w RL (poważny) `network.py:16`

```python
self.bn = nn.BatchNorm1d(conv_filters)  # ← problematyczne w RL
```

BatchNorm ma dwa tryby:
- `train()` → liczy mean/std z **aktualnego batcha**
- `eval()` → używa **running statistics** (akumulowane przez trening)

W każdym kroku Double DQN następuje toggle `eval()` → `train()` (`trainer.py:449–451`):

```python
self.main_network.eval()
next_q_main = self.main_network(next_state_tensors, action_mask=next_masks)
self.main_network.train()
```

Podczas `eval()` BN używa running stats z poprzednich iteracji — które mogą być nieaktualne, bo rozkład danych ciągle się zmienia wraz z polityką agenta. To powoduje niespójność statystyk między selekcją akcji (eval) a właściwym treningiem (train).

### Fix

Zastąpić `BatchNorm1d` przez `LayerNorm` w `Conv1DBlock`:

```python
# Zamiast:
self.bn = nn.BatchNorm1d(conv_filters)

# Użyć:
self.ln = nn.LayerNorm(conv_filters)

# W forward():
# Zamiast: x = self.bn(x)
# Użyć:    x = x.transpose(1, 2); x = self.ln(x); x = x.transpose(1, 2)
# lub po GAP (po x.mean(dim=2)):
x = self.ln(x)  # LayerNorm po [batch, filters]
```

`LayerNorm` nie ma trybów train/eval — normalizuje per-próbka, jest stabilny w RL.

---

## Problem 2 — Global Average Pooling niszczy sekwencję przed Transformerem `network.py:29`

```python
x = x.mean(dim=2)   # [batch, filters, seq_len] → [batch, filters]
```

Conv1D przetwarza np. 30 świec 1m i produkuje `[batch, 64, 30]`. Następnie **cała sekwencja jest uśredniana do jednego wektora** `[batch, 64]`. Ten wektor staje się jednym tokenem dla Transformera.

```
Transformer widzi tylko 4 tokeny:
  Token_1m  = mean_over_time([batch, 64])
  Token_15m = mean_over_time([batch, 64])
  Token_1h  = mean_over_time([batch, 64])
  Token_1d  = mean_over_time([batch, 64])
```

Self-attention po 4 tokenach to właściwie ważona suma 4 wektorów — siła Transformera (attention po długich sekwencjach) jest całkowicie zmarnowana. Sieć nie widzi żadnej struktury czasowej wewnątrz timeframe'u.

### Fix

Przekazać pełną sekwencję do Transformera zamiast agregować przed nim:

```python
# Zamiast GAP + stack:
tokens = torch.stack([self.projection(c) for c in conv_outputs], dim=1)
# [batch, num_timeframes, filters] — tylko 4 tokeny

# Użyć pełnej sekwencji z projekcją do d_model:
# conv_out shape: [batch, filters, seq_len] → transpose → [batch, seq_len, filters]
# concat po seq_len → [batch, total_seq_len, d_model]
# Transformer dostaje setki tokenów zamiast 4
```

To jest główna zmiana proponowana w architekturze **Standard+** (issue #27).

---

## Problem 3 — Zbędna projekcja 64→64 `network.py:97`

```python
self.projection = nn.Linear(self.conv_filters, self.conv_filters)  # 64→64
```

Liniowa projekcja z tej samej do tej samej wymiarowości, bez aktywacji nieliniowej. Dodaje 64×64 = 4096 parametrów które mogą co najwyżej nauczyć się rotacji/skalowania przestrzeni cech. Bez nieliniowości nie zwiększa expressiveness modelu w żaden istotny sposób.

### Fix

Usunąć projekcję lub zastąpić projekcją do właściwego `d_model` jeśli wymiarowość ma się zmienić:

```python
# Jeśli d_model != conv_filters:
self.projection = nn.Linear(self.conv_filters, d_model)  # sensowna projekcja

# Jeśli d_model == conv_filters:
# usunąć projekcję całkowicie, przekazywać conv_out bezpośrednio
```

---

## Podsumowanie

| Problem | Plik | Linia | Wpływ | Trudność |
|---------|------|-------|-------|----------|
| BatchNorm zamiast LayerNorm | `network.py` | 16 | Wysoki — niespójność train/eval w RL | Łatwa |
| GAP niszczy sekwencję przed Transformerem | `network.py` | 29 | Wysoki — Transformer widzi 4 tokeny zamiast setek | Średnia |
| Projekcja 64→64 bez aktywacji | `network.py` | 97 | Niski — zbędne parametry | Łatwa |
